src/Accounts/Authentication/Extensions/ClientFactoryExtension.cs

```diff
@@ -0,0 +1,121 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Rest.TransientFaultHandling;
+using System;
+
+namespace Microsoft.Azure.Commands.Common.Authentication.Abstractions
+{
+    static class ServiceClientExtension
+    {
+        class HttpRetryTimes
+        {
+            static public int? AZURE_PS_HTTP_MAX_RETRIES
+            {
+                get
+                {
+                    return TryGetAzurePsHttpMaxRetries();
+                }
+            }
+            static public int? AZURE_PS_HTTP_MAX_RETRIES_FOR_429
+            {
+                get
+                {
+                    return TryGetAzurePsHttpMaxRetriesFor429();
+                }
+            }
+
+            private delegate string GetEnvironmentVariable();
+
+            static private int? TryGetValue(GetEnvironmentVariable GetEnvVariable)
```

Comment:
IMO, there is no need to use delegate here as it does not improve code readability . How about use TryGetValue(string environmentName)?